java/src/main/java/org/whispersystems/libsignal/SignalProtocolAddress.java

```diff
@@ -34,11 +47,15 @@ public boolean equals(Object other) {
     if (!(other instanceof SignalProtocolAddress)) return false;
 
     SignalProtocolAddress that = (SignalProtocolAddress)other;
-    return this.name.equals(that.name) && this.deviceId == that.deviceId;
+    return this.getName().equals(that.getName()) && this.getDeviceId() == that.getDeviceId();
   }
 
   @Override
   public int hashCode() {
-    return this.name.hashCode() ^ this.deviceId;
+    return toString().hashCode();
```

Comment:
No longer utilizing deviceId in hash?

Reply:
The `toString` output includes the name and deviceId so while the hash changes it is functionally identical. Maybe it is better for review purposes to replace this with `getName().hashCode() ^ getDeviceId().hashCode()`. That also would prevent the hash ordering from changing, though I would hope nothing is expecting that to be stable across different binaries.

Reply:
Fixed as described above. An interesting and perhaps unintentional aspect of this hash code definition is it has high locality on the sender name. Eg for some given name, H(name,deviceId=0) and H(name,deviceId=1) are just 1 apart. That may or may not be beneficial, not sure.

Reply:
> That also would prevent the hash ordering from changing, though I would hope nothing is expecting that to be stable across different binaries.

If folks care about the order they insert items into a Map or Set, they can use the appropriate subclass :) (LinkedHashMap or LinkedHashSet, for example)